README.rst

```diff
@@ -213,7 +213,24 @@ This CMake macro simplifies the CMake package configure and install by performin
 
 .. code-block:: cmake
 
-   configure_package(NAMESPACE <PACKAGE_NAMESPACE> TARGETS <TARGET_NAME_A> <TARGET_NAME_B>)
+   configure_package(
+     NAMESPACE <PACKAGE_NAMESPACE>
+     TARGETS <TARGET_NAME_A> <TARGET_NAME_B>
+     DEFAULT_TEMPLATE | CONFIG_TEMPLATE <Template file path>
+     DEPENDENCIES <deps>...
+     CFG_EXTRAS <cmake files>...
+   )
+
+Note the use of either `DEFAULT_TEMPLATE` or `CONFIG_TEMPLATE` for describing the `*-config.cmake`
```

Comment:
Is there a reason to not make the DEFAULT_TEMPLATE an option where if it is not enable the default is to expect a config template instead of having to provide the config template path?

Reply:
My intention was to move towards the default being the autogenerated file. So it deprecates the default behavior of expecting a file present in favor of having to explicitly specify it. Sometime later we would then remove the `DEFAULT_TEMPLATE` option and it would be the new default behavior.

Reply:
For backward compatibility what if it is changed to check if a local config exists and if not it uses the default template? Then the DEFAULT_TEMPLATE and CONFIG_TEMPLATE can be removed. Also a check can be added for if you provide DEPENDENCIES or EXTRA_CFG and a local config was found print an error that this is only supported when using the default template.

Reply:
That sounds reasonable to avoid a deprecation.

Reply:
Done.